src/packages/cli/src/Studio.ts

```diff
@@ -0,0 +1,162 @@
+      ${chalk.dim('$')} BROWSER=firefox prisma studio --port 5555 --experimental
+
+      Start Studio without opening in a browser
+      ${chalk.dim('$')} prisma studio --port 5555 --browser none --experimental
+      ${chalk.dim('$')} BROWSER=none prisma studio --port 5555 --experimental
+  `)
+
+  private constructor(private readonly providerAliases: ProviderAliases) {
+    this.providerAliases = providerAliases
+  }
+
+  /**
+   * Parses arguments passed to this command, and starts Studio
+   *
+   * @param argv Array of all arguments
+   */
+  public async parse(argv: string[]): Promise<string | Error> {
+    const args = arg(argv, {
+      '--help': Boolean,
+      '-h': '--help',
+      '--port': Number,
+      '-p': '--port',
+      '--browser': String,
+      '-b': '--browser',
+      '--schema': String,
+      '--experimental': Boolean,
+    })
+
+    if (isError(args)) {
+      return this.help(args.message)
+    }
+
+    if (args['--help']) {
+      return this.help()
+    }
+
+    if (!args['--experimental']) {
+      throw new ExperimentalFlagError()
+    }
+
+    const schemaPath = getSchemaPathSync(args['--schema'])
+    if (!schemaPath) {
+      throw new Error(`Could not find ${args['--schema'] || 'schema.prisma'}`)
+    }
+
+    const port =
+      args['--port'] || (await getPort({ port: getPort.makeRange(5555, 5600) }))
+    const browser = args['--browser'] || process.env.BROWSER
+    const platform = await getPlatform()
+    const extension = platform === 'windows' ? '.exe' : ''
+    const queryEnginePath =
+      process.env.NODE_ENV === 'production'
+        ? eval(
+            `require('path').join(__dirname, '../query-engine-${platform}${extension}')`,
+          )
+        : eval(
+            `require('path').join(__dirname, '../node_modules/@prisma/sdk/query-engine-${platform}${extension}')`,
+          )
+    const staticAssetDir =
+      process.env.NODE_ENV === 'production'
+        ? path.resolve(__dirname, './public')
+        : path.resolve(__dirname, '../dist/public')
+
+    const studio = new StudioServer({
+      schemaPath,
+      port,
+      prismaClient: {
+        generator: {
+          version: packageJson.prisma.version,
+          providerAliases: this.providerAliases,
+        },
+      },
+      binaryPaths: {
+        queryEngine: queryEnginePath,
+      },
+      staticAssetDir,
+      versions: {
+        prisma2: packageJson.version,
+        queryEngine: packageJson.prisma.version,
+      },
+    })
+
+    await studio.start()
+
+    const serverUrl = `http://localhost:${port}`
+    switch (browser) {
+      case 'none':
+      case 'NONE':
+        break
+      default:
+        await open(serverUrl, {
+          app: browser,
+          url: true,
+          // eslint-disable-next-line @typescript-eslint/no-empty-function
+        }).catch(() => {}) // Ignore any errors
+        break
+    }
+
+    return `Studio started at ${serverUrl}`
```

Comment:
Any errors from here will automatically bubble to stdout right? So I do not need to wrap everything in the `parse` function in a `try..catch`?

Reply:
Indeed not sure why there was one before.

Reply:
You might want to use a simple `if(browser.toLowerCase() === 'none')` instead, the switch is a bit more work to read.

Reply:
It's nicer indeed!